src/hooks/locations/useFetchLocations.ts

```diff
@@ -0,0 +1,48 @@
+import axios, { CancelTokenSource } from 'axios';
+import { useCallback, useEffect, useState } from 'react';
+
+import { ERROR_REQUEST_CANCELLED_BY_CLIENT } from '../../constants/api.constants';
+import { ILocation } from '../../interfaces/location.interface';
+import { locationService } from '../../services/location.service';
+import log from '../../utils/log.utils';
+
+type FetchResult = {
+  locations: ILocation[] | undefined;
+  isLoading: boolean;
+  error: string | null;
+};
+
+const useFetchLocations = (): FetchResult => {
```

Comment:
**suggestion:** a lot of this hooks are pretty similar, you could have a look how to generalize them in 1 or 2 variants

Reply:
Yep, great idea in fact 😅 Working on it now